pypesto/hierarchical/solver.py

```diff
@@ -298,3 +296,74 @@ def fun(x):
             x_opt = scale_value_dict(x_opt, problem)
 
         return x_opt
+
+    def sample_startpoints(
+        self, problem: InnerProblem, pars: List[InnerParameter]
+    ) -> np.ndarray:
+        """Sample startpoints for the numerical optimization.
+
+        Samples the startpoints for the numerical optimization from a
+        log-uniform distribution using the symmetric logarithmic scale.
+
+        Parameters
+        ----------
+        problem:
+            The inner problem to solve.
+        pars:
+            The inner parameters to sample startpoints for.
+
+        Returns
+        -------
+        The sampled startpoints appended to the cached startpoints.
+        """
+        if self.minimize_kwargs['n_starts'] == 1 and self.x_guesses is None:
+            return np.array(
+                [list(problem.get_dummy_values(scaled=False).values())]
+            )
+        elif self.x_guesses is not None:
+            n_samples = self.minimize_kwargs['n_starts'] - len(self.x_guesses)
+        else:
+            n_samples = self.minimize_kwargs['n_starts'] - 1
+
+        if n_samples <= 0:
+            return self.x_guesses
+
+        lb = np.array(
+            [x.lb if x.lb != -np.inf else self.dummy_lb for x in pars]
+        )
+        ub = np.array(
+            [x.ub if x.ub != np.inf else self.dummy_ub for x in pars]
+        )
```

Comment:
Will replace `nan`s as well (`nan` -> `0`), but I guess the bounds have already been checked to not contain `nan` at this point, since it would be an issue for either implementation.

```suggestion
        lb = np.nan_to_num([x.lb for x in pars], neginf=self.dummy_lb)
        ub = np.nan_to_num([x.ub for x in pars], posinf=self.dummy_ub)
```

Reply:
Hmmm probably, but it's good to be safe. Nice ty